packages/react/src/provider/use-context-mutator.ts

```diff
@@ -0,0 +1,24 @@
+import { useContext } from 'react';
+import { OpenFeature, EvaluationContext } from '@openfeature/web-sdk';
+import { Context } from './context';
+
+/**
+ *
+ * A hook for accessing context mutating functions.
+ *
+ */
+export function useContextMutator() {
+  async function mutateContext(updatedContext: EvaluationContext): Promise<void> {
+    const { domain } = useContext(Context) || {};
+
+    if (!domain) {
+      throw new Error('No domain set for your context');
+    }
+
+    OpenFeature.setContext(domain, updatedContext);
+  }
+
+  return {
+    mutateContext,
```

Comment:
I saw the comment about merging contexts, I think I need to know more about use cases in the real world. For example I saw one project setting the global context instead of a domain scoped one in react, is that also a valid use case? Should we just stick with updating the context for a domain for now and expand on this hook later?

Reply:
Yep, I think it makes sense to add later. This is already a great start. I just wonder if we should name this function "setContext" instead of "mutateContext". @toddbaert do you have an opinion?